prow/plugins/blunderbuss/blunderbuss.go

```diff
@@ -29,12 +29,19 @@ import (
 	"k8s.io/test-infra/prow/pluginhelp"
 	"k8s.io/test-infra/prow/plugins"
 	"k8s.io/test-infra/prow/plugins/assign"
+	"regexp"
 )
 
 const (
 	pluginName = "blunderbuss"
 )
 
+var (
+	autoAssignRe    = regexp.MustCompile(`(?mi)^/assign-reviewers$`)
+	noAutoAssignRe  = regexp.MustCompile(`(?mi)^/no-assign-reviewers$`)
+	noAssignTitleRe = regexp.MustCompile(`(?i)^\W?WIP\W`)
```

Comment:
Will this match if `WIP` is the first or last word of the title? If it's Perl-style regex, can use `\b` for a word boundary (instead of `\W`)?

Reply:
Also, do you want that `^` in there?

Reply:
So this is the same regex as used in the `wip` plugin, and I'd rather keep behaviour consistent as it otherwise just adds more divergence from upstream and magic behaviour for people to get confused by. It's been pretty reliable at detecting WIP PRs in the past, though.

I'm intending on extracting this in another commit in a moment, so that there's not any duplication.

Reply:
Ah I missed the `?`.

So it's basically requiring `WIP` to be at the beginning, fair enough.

Reply:
It may be worth trying to upstream a change to remove the `^` anchor though, to allow PR titles like `A-1 WIP ...`, but I'd rather see if this causes any trouble before doing so

Reply:
Yeah, would suggest `(?i)\bWIP\b` if you want to catch all of the typical formats we would use...

Reply:
How would you feel about `(?i)(^\W?WIP\W|\[WIP\]|\(WIP\))`? It seems that with `\b` it won't match "[WIP] ..." as the `\b` doesn't seem to recognise the initial word boundary. This one matches as per https://regex101.com/r/MhPwNx/1

I could only find one false negative (where the PR opener had written `[WIP - DO NOT MERGE]`, which is now automated with a `wip/do-not-merge` label) for this in the last 4 pages of platform marges matching `WIP` but there was one false positive for just removing the `^` anchor.

I can see if they'd like the behaviour upstream at a later point - any opinions?

Reply:
`(?i)\bWIP\b` works on all those examples for me?

Reply:
Oh you mean if you do something like `\b\[WIP` - that won't work, because between space and `[` is not a word boundary.

Reply:
Oh, so it does -- I must've made a typo; sorry!

Any opinions regarding `(?i)(\bWIP\b|\[WIP\]|\(WIP\))` vs `(?i)\bWIP\b`? I'd rather avoid the false negatives from the latter but there's unlikely to be many so I'm open to either.

Reply:
Ah, no, it's `^\bWIP` which I was trying, which was causing issue - so it's between `(?i)(^\W?WIP\W|\[WIP\]|\(WIP\))` and `(?i)\bWIP\b`.

I'm starting to be inclined to the latter just for maintainability of this code.

Reply:
There shouldn't be any difference between those two.

For the one you said earlier, maybe something like `WRK-123: WIP stuff` would be an issue.

Reply:
Out of sync messages. For your latest message, I prefer the latter due to `WRK-123: WIP stuff`?

Reply:
After discussion: `(?i)(^\W?WIP\b|\[WIP\]|\(WIP\))`